lib/commands/deploy/index.js

```diff
@@ -159,5 +157,34 @@ with build flow ${codeProperty.buildFlow}.`);
     });
 }
 
+function _deploySkillMetadata(options, callback) {
+    const { profile, target } = options;
+    if (!target || target === CONSTANTS.DEPLOY_TARGET.SKILL_METADATA) {
+        // Skill Metadata
+        Messenger.getInstance().info('==================== Deploy Skill Metadata ====================');
+        const skillMetaSpinner = new SpinnerView();
+        skillMetaSpinner.start('Uploading the entire skill package (it may take few minutes to build the skill metadata)...');
+        helper.deploySkillMetadata(options, (metaErr) => {
+            skillMetaSpinner.terminate();
+            if (metaErr && metaErr !== 'The hash of current skill package folder does not change compared to the last deploy hash result, '
+        + 'CLI will skip the deploy of skill package.') {
+                return callback(metaErr);
+            }
+            if (metaErr) {
+                // this case is the warning message of the same hash skip, deploy will continue
+                Messenger.getInstance().warn(metaErr);
+            } else {
+                ResourcesConfig.getInstance().write();
+                Messenger.getInstance().info('Skill package deployed successfully.');
+            }
+            Messenger.getInstance().info(`Skill ID: ${ResourcesConfig.getInstance().getSkillId(profile)}`);
+
+            return callback();
+        });
+    } else {
+        return callback();
+    }
```

Comment:
i suggest we do the callback first with the predicate that target doesn't include skill_metadata. Return with a short callback, and avoid adding additional indent.